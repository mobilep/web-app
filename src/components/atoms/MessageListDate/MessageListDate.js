import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { format, isToday, isYesterday } from 'date-fns';

const getClassName = (className) => classNames('LineOnSides', className);

const getDateString = (timestamp, languageCode, yesterday, today) => {
	let relativeDayString = '';
	if (isToday(timestamp)) {
		relativeDayString = `${today} `;
	} else if (isYesterday(timestamp)) {
		relativeDayString = `${yesterday} `;
	}

	return `${relativeDayString}${format(timestamp, 'd MMMM', { locale: languageCode })}`;
};

const MessageListDate = ({ className, languageCode, timestamp, yesterdayTextLabel, todayTextLabel }) => {
	return (
		<div className={getClassName(className)}>
			{getDateString(timestamp, languageCode, yesterdayTextLabel, todayTextLabel)}
		</div>
	);
};

MessageListDate.propTypes = {
	className: PropTypes.string,
	languageCode: PropTypes.object,
	timestamp: PropTypes.number,
	todayTextLabel: PropTypes.string,
	yesterdayTextLabel: PropTypes.string,
};

export default MessageListDate;
