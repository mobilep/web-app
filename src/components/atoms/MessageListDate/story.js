import React from 'react';
import { storiesOf } from '@storybook/react';
import { number } from '@storybook/addon-knobs/react';

import MessageListDate from './MessageListDate';

const timestamp = new Date().getTime();

storiesOf('1. Atoms / MessageListDate', module)
	.add('default', () => (
		<MessageListDate
			timestamp={number('Timestamp', timestamp)}
		/>
	));
