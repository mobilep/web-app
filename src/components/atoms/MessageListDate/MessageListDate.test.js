import React from 'react';
import { shallow } from 'enzyme';

import MessageListDate from './MessageListDate';

describe('Atom - MessageListDate', () => {

	it('Should render', () => {
		const wrapper = shallow(<MessageListDate timestamp={1530708467971} />);
		expect(wrapper).toMatchSnapshot();
	});

});

