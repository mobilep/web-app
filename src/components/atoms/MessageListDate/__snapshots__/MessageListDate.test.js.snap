// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - MessageListDate Should render 1`] = `
<div
  className="LineOnSides"
>
  4 July
</div>
`;
